metrics/metrics.go

```diff
@@ -84,21 +83,20 @@ func RegisterRootScope(rep ScopeFunc) {
 		panic("There can be only one metrics reporter")
 	}
 
-	_scopeFunc = rep
+	_scopeFunc = scopeFunc
 }
 
-// RootScope returns the provided stats reporter, or nil
-func RootScope(i ScopeInit) (tally.Scope, io.Closer) {
+// RootScope returns the provided metrics scope and stats reporter, or nil if not provided
+func RootScope(i ScopeInit) (tally.Scope, tally.StatsReporter, io.Closer) {
 	_mu.Lock()
 	defer _mu.Unlock()
 
 	if _scopeFunc != nil {
-		rep, closer, err := _scopeFunc(i)
+		scope, reporter, closer, err := _scopeFunc(i)
 		if err != nil {
 			panic(fmt.Sprintf("Failed to initialize metrics reporter %v", err))
 		}
-		return rep, closer
+		return scope, reporter, closer
 	}
-
-	return nil, nil
+	return nil, nil, nil
```

Comment:
Well, I am talking about dealing with individual return parameters is strange for the caller. Right now you have to check for each return variable in service.go and assign null scope, add check for each nil/not nil, and so on. encapsulating what rootScope returns would make the code concise and readable. 

For error I am not talking about panic, but instead of returning nil, nil, nil, it would make more sense for the caller to simply check for err message (something along the lines that no scope func is registered) 
The method would look like this - 
`
func RootScope(i ScopeInit) (scopeRegistry, error) {
  if _scopeFunc != nil {
    return scopeRegistry{}, nil
  }
  return nil, errors.New("no scope functions registered")
}
`